tests/unit/commands/validate/lib/test_sam_template_validator.py

```diff
@@ -306,3 +413,131 @@ def test_replace_local_codeuri_with_no_resources(self):
         # check template
         self.assertEqual(validator.sam_template.get("Resources"), {})
+
+    def test_DefinitionBody_gets_replaced_in_api(self):
+        template = {
+            "AWSTemplateFormatVersion": "2010-09-09",
+            "Transform": "AWS::Serverless-2016-10-31",
+            "Resources": {
+                "ServerlessFunction": {"Type": "AWS::Serverless::Function"},
+                "ServerlessApi": {
+                    "Type": "AWS::Serverless::Api",
+                    "Properties": {
+                        "StageName": "Prod",
+                        "DefinitionBody": {
+                            "Fn::Transform": {
+                                "Name": "AWS::Include",
+                                "Parameters": {"Location": "./tests/unit/commands/validate/lib/openapi/openapi.yaml"},
+                            }
+                        },
+                    },
+                },
+            },
+        }
+
+        managed_policy_mock = Mock()
+
+        validator = SamTemplateValidator(template, managed_policy_mock)
+
+        validator._replace_local_openapi()
+
+        template_resources = validator.sam_template.get("Resources")
+        self.assertIn("DefinitionBody", template_resources.get("ServerlessApi").get("Properties"))
+        self.assertNotIn(
+            "Fn::Transform", template_resources.get("ServerlessApi").get("Properties").get("DefinitionBody")
+        )
+        self.assertIn("openapi", template_resources.get("ServerlessApi").get("Properties").get("DefinitionBody"))
+
+    def test_DefinitionBody_not_replaced_if_file_not_found(self):
+        template = {
+            "AWSTemplateFormatVersion": "2010-09-09",
+            "Transform": "AWS::Serverless-2016-10-31",
+            "Resources": {
+                "ServerlessApi": {
+                    "Type": "AWS::Serverless::HttpApi",
+                    "Properties": {
+                        "StageName": "Prod",
+                        "DefinitionBody": {
+                            "Fn::Transform": {
+                                "Name": "AWS::Include",
+                                "Parameters": {"Location": "./tests/unit/commands/validate/lib/openapi/notafile.yaml"},
+                            }
+                        },
+                    },
+                }
+            },
+        }
+
+        managed_policy_mock = Mock()
+
+        validator = SamTemplateValidator(template, managed_policy_mock)
+
+        validator._replace_local_openapi()
+
+        template_resources = validator.sam_template.get("Resources")
+        self.assertIn("DefinitionBody", template_resources.get("ServerlessApi").get("Properties"))
+        self.assertNotIn("openapi", template_resources.get("ServerlessApi").get("Properties").get("DefinitionBody"))
+        self.assertIn("Fn::Transform", template_resources.get("ServerlessApi").get("Properties").get("DefinitionBody"))
+
+    def test_DefinitionBody_gets_replaced_if_json(self):
```

Comment:
For the tests that read and touch the file system, can we move these to `integ` tests? 

Generally, the unit tests through the repo utilize mocks and (try not to) touch the system at all. Instead, integ tests are preferred for the end to end. This will also allow us to test the functionality from the CLI interface. 

I don't mind having these but would like to have some basic cases covered with integ tests so we can validate the end to end experience and gives us a good guard for functionality we support, in the case of future regression.

Reply:
I agree they should be mocked (still learning Python mocking!).
Will rework the tests with mocks and add the none mocked versions into integ tests.

Reply:
@NickHeap2 Let me know if you need some help in the mocking area. Happy to lend a hand to help if you get stuck at all.

Reply:
@jfuss The part that took me a while was getting the patch path for parse_yaml file correct.
I've finished changing the unit tests to use mocks and added integration test cases now.